clarity_ext/domain/common.py

```diff
@@ -2,36 +2,18 @@
 import copy
 
 
-class DomainObjectMixin(object):
+class DomainObject(object):
+    def __init__(self, id):
+        self.id = id
 
     def __eq__(self, other):
         if isinstance(other, self.__class__):
             return self._eq_rec(self, other)
         else:
             return False
 
-    def __hash__(self):
-        return hash(self.id)
-
     def __lt__(self, other):
-        return self.compare(other) < 0
-
-    def __gt__(self, other):
-        return self.compare(other) > 0
-
-    def __le__(self, other):
-        return self.compare(other) <= 0
-
-    def __ge__(self, other):
-        return self.compare(other) >= 0
-
-    def compare(self, other):
-        # Override if needed
-        if self._eq_rec(self, other) == 0:
-            return 0
-        elif str(self) < str(other):
-            return -1
-        return 1
+        return self.id < other.id
```

Comment:
Is `__lt__` tested for when a list contains None values?

Reply:
Excellent catch :)

Reply:
I think that it's perhaps OK if one gets an error if trying to sort an object with incompatible objects or None. One can always fix it by explicitly providing a key function:

```sorted(lst, key=lambda a: a.id if a else 0)```